### PR TITLE
[WIP] Update FAQ: Min system requirements

### PR DESCRIPTION
**Pull request by dsarkar**

### CODE REVIEW

**Comment by dsarkar** [comment]
image

**Comment by dsarkar** [comment]
EN follows .... stand by

**Comment by Ein-Tim** [comment]
This is all about iOS, I thought again about it and maybe it would be better placed in this FAQ entry: https://www.coronawarn.app/de/faq/#ios135?

**Comment by dsarkar** [comment]
EN

UPDATE 8 February 2021  
As of February 5, 2021, only the Corona-Warn-App (CWA) versions 1.5 and later are supported. For this reason, users with older versions will receive a notification to update their app accordingly. This applies to both iOS and Android. At the same time, however, certain minimum requirements apply to the operating system: Android at least 6.0 and iOS at least 13.7.  
iOS users who have previously used iOS 13.5, for example, must now update the operating system to at least iOS 13.7 in order to continue using the CWA. Only then the technically correct function of the app be ensured.  
Please update your iOS version first. Older smartphones, such as the iPhone SE, 6S or 6S Plus, also support the latest iOS operating system. This allows you to upgrade your iPhone to at least iOS 13.7. For security reasons, however, you should keep your iPhone up-to-date, as Apple also regularly fixes minor security vulnerabilities with the updates. You can do this on your iPhone under Settings - General - Software Update.  
Subsequently, please update the Corona-Warn-App. The Corona-Warn-App must also be installed at least in version 1.5. Version 1.11 is already available.  
Starting with CWA version 1.12, the older iPhone models iPhone 5s, iPhone 6, iPhone 6 Plus are also supported. These devices must be upgraded to iOS version 12.5 or 12.5.1.  
 
The following table is for an overview:

**Comment by dsarkar** [comment]
@Ein-Tim hm, would aso be a place where to publish it, however it also mentions "Android". Maybe a link in https://www.coronawarn.app/de/faq/#ios135 to this article about minimum requirements in general.

**Comment by Ein-Tim** [comment]
@dsarkar 

Yes, please include a link in https://www.coronawarn.app/de/faq/#ios135 👍

**Comment by dsarkar** [comment]
EN

image

**Comment by dsarkar** [comment]
DE

image

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:328]: Ihre App => ihre App (es sind die Apps der Nutzer, die hier angesprochen werden)
Android mindestens => für Android, mindestens (hört sich besser an)

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:329]: Ihr iOS Betriebsystem => ihr iOS Betriebssystem (Grammar and Typo)

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:330]: sollten Sie ihr iPhone => sollten Sie Ihr iPhone (Grammar)

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json:331]: iPhone 5s, iPhone 6, iPhone 6plus => iPhone 5s, iPhone 6 bzw. iPhone 6plus
Diese Geräte müssen auf die iOS Version 12.5 bzw. 12.5.1 aktualisiert werden (Full stop missing at end of sentence)

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json]: neuste Version => neueste Version 
(in the previous row it says "neuere Modelle", so keep to the same from of neu...)

File: src/data/faq_de.json
Comment (MikeMcC399) [inline src/data/faq_de.json]: AKTUELL => AKTUELL 4. Februar 2021
since there is another AKTUELL at the top of this article "AKTUELL 8. Februar 2021"

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Here are quite a few suggestions for correction for you, for the German article.

After Version 1.12 is released this article should be simplified to remove some of the future statements.

File: src/data/faq_de.json
Comment (dsarkar) [reply src/data/faq_de.json]: ok

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:321]: Only then the technically correct operation of the app can be ensured.
=>
Only then can the technically correct operation of the app be ensured. (Word order)

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:323]: iPhone 5s, iPhone 6, iPhone 6 Plus => iPhone 5s, iPhone 6 and iPhone 6 Plus

File: src/data/faq.json
Comment (MikeMcC399) [inline src/data/faq.json:324]: is for => provides

File: src/data/faq_de.json
Comment (dsarkar) [reply src/data/faq_de.json]: ok

File: src/data/faq_de.json
Comment (dsarkar) [reply src/data/faq_de.json:331]: ok

**Review by MikeMcC399 — CHANGES_REQUESTED** [review]
Now the comments for the English article.

File: src/data/faq_de.json
Comment (dsarkar) [reply src/data/faq_de.json]: @MikeMcC399 I removed the previous "AKTUELL" it since it was only the link to the blog, so no historic information is lost in this case.

File: src/data/faq.json
Comment (dsarkar) [reply src/data/faq.json:321]: ok

File: src/data/faq.json
Comment (dsarkar) [reply src/data/faq.json:324]: ok

File: src/data/faq_de.json
Comment (dsarkar) [reply src/data/faq_de.json:329]: ok

File: src/data/faq_de.json
Comment (dsarkar) [reply src/data/faq_de.json]: ok

File: src/data/faq.json
Comment (dsarkar) [reply src/data/faq.json:323]: ok

**Comment by dsarkar** [comment]
@MikeMcC399 Many many thanks!

**Comment by MikeMcC399** [comment]
@dsarkar 
I just skimmed through the changed articles and they now look good. 👍 

The only open point is about the forced update on Android (currently forced update to 1.0.4, not 1.5). 

Of course it is quite possible to have a support policy which says the minimum supported level is 1.5 for Android (and iOS) and that this policy is not strictly enforced through a forced update. It just means that if a user reports a problem with a version less than 1.5 then the response will be to please update first, then come back if the problem still occurs.

**Comment by dsarkar** [comment]
@MikeMcC399 
Thanks again for the incredible thorough proofreading. 

> The only open point is about the forced update on Android (currently forced update to 1.0.4, not 1.5).

Yes, we are waiting for some feedback. My understanding is that in any case, in the very near future it will be min req. 1.5.1 for Android and 1.5.3 for iOS.

**Comment by MikeMcC399** [comment]
@dsarkar 
> Thanks again for the incredible thorough proofreading.

You are very welcome! 🙂 Thank you also for incorporating the changes I suggested!

> > The only open point is about the forced update on Android (currently forced update to 1.0.4, not 1.5).
> 
> Yes, we are waiting for some feedback. My understanding is that in any case, in the very near future it will be min req. 1.5.1 for Android and 1.5.3 for iOS.

That would make a lot of sense.

**Comment by MikeMcC399** [comment]
Unfortunately the tables do not render well on a mobile device and this information probably needs to be displayed in a different way.

You can see what it looks like on a mobile device if you reduce the browser on a desktop / laptop to its minimum width.

BTW: This also affects the table https://www.coronawarn.app/en/faq/#interoperability_countries

**Comment by MikeMcC399** [comment]
@dsarkar 
To optimize the display on the browser of a mobile device the table needs to be transformed into plain text. The table is just too wide for a mobile device. Do you agree? Do you want me to propose some text for this?

Also I assume that this PR will not be merged until after 1.12 is released. Is the release still planned for tomorrow Feb 10, 2021?

**Comment by dsarkar** [comment]
@MikeMcC399 

> To optimize the display on the browser of a mobile device the table needs to be transformed into plain text. The table is just too wide for a mobile device. Do you agree? Do you want me to propose some text for this?

Yes, I did not check, but I thought that we might run into a problem there. If you want to propose a text a will be happily integrating this into the FAQ. Thanks! 

> Also I assume that this PR will not be merged until after 1.12 is released. Is the release still planned for tomorrow Feb 10, 2021?

The blog post says so at least. https://www.coronawarn.app/en/blog/2021-02-04-cwa-1-12-announcement/

**Comment by dsarkar** [comment]
@MikeMcC399 Just got feedback. Min req. version CWA
iOS 1.5.3
Android 1.0.4

**Comment by dsarkar** [comment]
@MikeMcC399 

here on GitHub no impact visible. However, in AppStore, there were a few complaints on iOS, and apparently via email more about Android. Based on the impact it was decided for the moment:

iOS 1.5.3
Android 1.0.4

**Comment by MikeMcC399** [comment]
@dsarkar 

Here is a suggestion to replace the table. It is not a 1:1 replacement, however it attempts to summarize the important information. Whilst I was writing this, I thought it would be good to update and simplify en/faq/#ENF_version so I will submit an issue, then a PR for this.

---

en/faq/#minimum_requirements

**iOS - Minimum operating system requirements**

The minimum operating system level for iPhone 5s, iPhone 6 and iPhone 6 Plus devices is iOS 12.5. For all other iOS devices the minimum is iOS 13.7. These versions of the iOS operating system include the necessary v2 Apple Exposure Notification Framework.

**Android - Minimum operating system requirements**

The minimum operating system level for all Android devices is version 6 of Android. In addition, the device must have Google Play services installed, with a minimum version 1.5 of the Google Exposure Notification System (ENS). See en/faq/#ENF_version for instructions about how to find out which version of ENS is installed.

----
de/faq/#minimum_requirements

**iOS - Mindestvoraussetzungen für das Betriebssystem**

Für die Geräte iPhone 5s, iPhone 6 bzw. iPhone 6 Plus wird mindestens iOS Version 12.5 benötigt. Alle anderen iOS-Geräte benötigen mindestens iOS Version 13.7. Diese Versionen des iOS-Betriebssystems beinhalten auch die benötigte v2 Apple Exposure Notification Framework.

**Android - Mindestvoraussetzungen für das Betriebssystem**

Alle Android-Geräte benötigen mindestens Version 6 (Marshmallow) des Android-Betriebssystems. Zusätzlich muss das Gerät die Google Play-Dienste mit einer Mindestversion 1.5 des Google COVID-19-Benachrichtigungssystems installiert haben. Lesen Sie de/faq/#ENF_version um Informationen zu bekommen, wie man die installierte Version des COVID-19-Benachrichtigungssystems feststellt.

**Comment by MikeMcC399** [comment]
@dsarkar 
>here on GitHub no impact visible. However, in AppStore, there were a few complaints on iOS, and apparently via email more about Android. Based on the impact it was decided for the moment:
>
>iOS 1.5.3
>Android 1.0.4

Many thanks for this information! Do we know what problems were caused by setting a higher minimum version?

**Comment by dsarkar** [comment]
@MikeMcC399 Thanks. LGTM. The table is nicer, but not appropriate for mobile devices.

**Comment by dsarkar** [comment]
> Many thanks for this information! Do we know what problems were caused by setting a higher minimum version?

@MikeMcC399 Regarding Android: My understanding is that there was no real technical problem, but that some users prefer for some reason not to update to higher versions.

**Comment by dsarkar** [comment]
@MikeMcC399 Actually I just learned there was a problem (I dont know the nature of the problem), will be fixed. And then min Android req will be 1.5.1

**Comment by MikeMcC399** [comment]
@dsarkar 
I have just submitted #865 to update the information about the Google Exposure Notifications System (ENS). This is just for completeness of the reference information. In fact these days there are no general issues about the version of ENS. If the device has an Internet connection then it will automatically get updated to the latest version of ENS because Google carries out a silent install / update of Google Play services. The last time I tried this out, after having done a factory reset on a device it took 7 seconds to get the latest version installed when connected via WiFi.

We're not seeing any major bugs in ENS any more so even if a device does not have the latest version, but only the previously released version, which is perhaps a few weeks older, there still should be no problem.

Will you go ahead with this PR now that CWA 1.12 has been announced as released (see https://www.coronawarn.app/en/blog/2021-02-10-corona-warn-app-version-1-12/)?

**Comment by dsarkar** [comment]
@MikeMcC399 

>  Diese Versionen des iOS-Betriebssystems beinhalten auch die benötigte v2 Apple Exposure Notification Framework.

- "die Framework", because of "die Arbeit" is a possibility how to derive the article, however, it seems to me that "das Framework" is more commonly used. What do you think?

**Comment by dsarkar** [comment]
- remove table
                             "The following table provides an overview:",
                            "<table class='table'><tr><th>Device</th><th>Operating System<br>Minimum</th><th>Operating System<br>Recommended</th><th>Exposure Notification System</th><th>CWA<br>Minimum</th><th>CWA<br>Preferred</th></tr><tr><td>iPhone 5s<br>iPhone 6<br>iPhone 6 Plus</td><td>iOS 12.5</td><td>iOS 12.5.1</td><td>v2 ENF (included in iOS)</td><td>Future<br>release<br>1.12</td><td>Future<br>release<br>1.12</td></tr><tr><td>iPhone SE<br>iPhone 6s<br>iPhone 6s Plus<br>and newer models</td><td>iOS 13.7</td><td>iOS 14.4</td><td>v2 ENF (included in iOS)</td><td>1.5</td><td>1.11</td></tr><tr><td>Android</td><td>Android 6</td><td>latest from device vendor</td><td>current Google Play services</td><td>1.5</td><td>1.11</td></tr></table>",

- add @MikeMcC399 suggestions. https://github.com/corona-warn-app/cwa-website/pull/852#issuecomment-775944636

**Comment by Ein-Tim** [comment]
@dsarkar "das Framework" sounds better

**Comment by MikeMcC399** [comment]
I think I probably meant to write the following, and shortened it too much:

... auch die benötigte Version v2 des Apple Exposure Notification Frameworks.

So also change the English version to match:

... necessary v2 version of the Apple Exposure Notification Framework.

**Comment by dsarkar** [comment]
rearrange
                           
 "As of February 5, 2021, only the Corona-Warn-App (CWA) versions 1.5 and later are supported. For this reason, users with older versions will receive a notification to update their app accordingly. This applies to both iOS and Android. At the same time, however, certain minimum requirements apply to the operating system: Android at least 6.0 and iOS at least 13.7.",
                           
 "iOS users who have previously used iOS 13.5, for example, must now update the iOS operating system to at least iOS 13.7 in order to continue using the CWA. Only then can the technically correct operation of the app be ensured.",
                           
 "<ul><li>Please update your iOS version first. Older smartphones, such as the iPhone SE, 6S or 6S Plus, also support the latest iOS operating system. This allows you to upgrade your iPhone to at least iOS 13.7. For security reasons, however, you should keep your iPhone up-to-date, as Apple also regularly fixes minor security vulnerabilities with the updates. You can do this on your iPhone under Settings - General - Software Update.</li><li>Subsequently, please update the Corona-Warn-App. The Corona-Warn-App must also be installed at least in version 1.5. Version 1.11 is already available.</li></ul>",
                            
"Starting with CWA version 1.12, the older iPhone models iPhone 5s, iPhone 6 and iPhone 6 Plus are also supported. These devices must be upgraded to iOS version 12.5 or 12.5.1.",

**Comment by dsarkar** [comment]
image

**Comment by dsarkar** [comment]
Android is not set to min 1.5.1, so I moved the part about forced system updates to iOS

**Comment by MikeMcC399** [comment]
@dsarkar 
write 1st generation instead of 1. generation for English text
https://en.wikipedia.org/wiki/IPhone_SE_(1st_generation)

This is a complex subject for iOS and the explanation is very readable and understandable.

For both iOS and Android the text looks good!

**Comment by dsarkar** [comment]
@Ein-Tim @MikeMcC399 thanks for the feedback and all the input.

I am having second thoughts now.

It might look just cleaner the following way:

- in the minimum_required article only @MikeMcC399 's text: https://github.com/corona-warn-app/cwa-website/pull/852#issuecomment-775944636
- and then as suggested earlier by @Ein-Tim move this part to the ios135 article:
image

**Comment by MikeMcC399** [comment]
@dsarkar 

Your idea to simplify and split sounds good.

I edited the changes I proposed into https://github.com/corona-warn-app/cwa-website/pull/852#issuecomment-775944636 to make it easier.

(When you finally merge this PR, could you please use squash and merge if possible, to keep the commits in the master branch to a minimum? I easily get lost in git! 😉 So that would help. See 
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-a-pull-request.)

**Comment by dsarkar** [comment]
move this to ios135
                            "As of February 5, 2021, only the Corona-Warn-App (CWA) versions 1.5.3 and later are supported. For this reason, users with older versions will receive a notification to update their app accordingly. iOS users who have previously used iOS 13.5 and CWA version 1.3, for example, must now update the iOS operating system to at least iOS 13.7 in order to continue using the CWA. Only then can the technically correct operation of the app be ensured.",
                            "<ul><li>Please update your iOS version first. Older smartphones, such as the iPhone SE, 6S or 6S Plus, also support the latest iOS operating system. This allows you to upgrade your iPhone to at least iOS 13.7. For security reasons, however, you should keep your iPhone up-to-date (currently iOS 14.4), as Apple also regularly fixes minor security vulnerabilities with the updates. You can do this on your iPhone under Settings - General - Software Update.</li><li>Subsequently, please update the Corona-Warn-App. The Corona-Warn-App must also be installed at least in version 1.5.3. The version 1.12.1 is already available.</li></ul><hr/>",

**Comment by MikeMcC399** [comment]
@dsarkar 
It would be good to get something published today to update
https://www.coronawarn.app/en/faq/#minimum_requirements
https://www.coronawarn.app/de/faq/#minimum_requirements

I just got a push notification from the [Bundesregierung App](https://play.google.com/store/apps/details?id=de.bundesregierung.bpaapp&hl=en) "Corona-Warn-App nun auch für ältere Smartphones"
https://www.bundesregierung.de/breg-de/such/update-cwa-1852448.

**Comment by dsarkar** [comment]
EN
image

**Comment by dsarkar** [comment]
removed:

                            "<br/>Seit dem 5. Februar 2021 werden nur die Corona-Warn-App (CWA) Versionen 1.5 und höher unterstützt. Aus diesem Grund erhalten Nutzer mit älteren Versionen einen Hinweis, ihre App entsprechend zu aktualisieren. Dies gilt sowohl für iOS als auch Android. Parallel gelten jedoch bestimmte Mindestvoraussetzungen an das Betriebssystem: Für Android mindestens 6.0 und iOS mindestens 13.7.",
                            "iOS Nutzer, die bisher bspw. iOS 13.5 genutzt haben, müssen ihr iOS Betriebssystem nun zwingend auf mindestens iOS 13.7 aktualisieren, um die CWA weiterhin zu nutzen. Nur in diesem Fall kann eine technisch korrekte Funktion der App sichergestellt werden.",       
                            "<ul><li>Bitte führen Sie zunächst ein Update Ihrer iOS-Version durch. Auch ältere Smartphones, wie bspw. das iPhone SE, 6S oder 6S Plus unterstützen das neueste iOS Betriebssystem. Sie können Ihr iPhone somit mindestens auf iOS 13.7 aktualisieren. Aus Sicherheitsgründen sollten Sie Ihr iPhone immer aktuell halten, da Apple auch regelmäßig kleinere Sicherheitslücken mit den Updates behebt. Sie können dies auf Ihrem iPhone unter Einstellungen – Allgemein – Softwareupdate durchführen.</li><li>Im Anschluss aktualisieren Sie bitte die Corona-Warn-App. Die Corona-Warn-App muss zudem mindestens in der Version 1.5 installiert sein. Inzwischen ist bereits Version 1.11 verfügbar. Wir bitten Sie daher, Ihre Corona-Warn-App zu aktualisieren.</li> </ul>",
                            "Ab der CWA Version 1.12 werden zusätzlich auch die ältere iPhone Modelle iPhone 5s, iPhone 6, bzw. iPhone 6 Plus unterstützt. Diese Geräte müssen auf die iOS Version 12.5 bzw. 12.5.1 aktualisiert werden.",
                            "Die folgende Tabelle dient zur Übersicht:",
                            "<table class='table'><tr><th>Gerät</th><th>Betriebssytem<br>Minimum</th><th>Betriebssytem<br>Empfohlen</th><th>Exposure Notification System</th><th>CWA<br>Minimum</th><th>CWA<br>Empfohlen</th></tr><tr><td>iPhone 5s<br>iPhone 6<br>iPhone 6 Plus</td><td>iOS 12.5</td><td>iOS 12.5.1</td><td>v2 ENF (inbegriffen in iOS)</td><td>Zukünftige<br>Version<br>1.12</td><td>Zukünftige<br>Version<br>1.12</td></tr><tr><td>iPhone SE<br>iPhone 6s<br>iPhone 6s Plus<br>und neuere Modelle</td><td>iOS 13.7</td><td>iOS 14.4</td><td>v2 ENF (inbegriffen in iOS)</td><td>1.5</td><td>1.11</td></tr><tr><td>Android</td><td>Android 6</td><td>neueste Version vom Gerätehersteller</td><td>aktuelle Google Play services</td><td>1.5</td><td>1.11</td></tr></table>",

**Comment by dsarkar** [comment]
DE

image

File: src/data/faq.json
Comment (Ein-Tim) [inline src/data/faq.json:321]: I would remove the v from the 2 since v stands for version. 

```suggestion
                            "The minimum operating system level for iPhone 5s, iPhone 6 and iPhone 6 Plus devices is iOS 12.5. For iPhone SE (1st generation), iPhone 6s, iPhone 6s Plus and newer iOS devices the minimum is iOS 13.7. These versions of the iOS operating system include the necessary version 2 of the Apple Exposure Notification Framework.",
```

File: src/data/faq_de.json
Comment (Ein-Tim) [inline src/data/faq_de.json:330]: Same as in English:

```suggestion
                            "Für die Geräte iPhone 5s, iPhone 6 bzw. iPhone 6 Plus wird mindestens iOS Version 12.5 benötigt. Das iPhone SE (1. Generation), iPhone 6s, iPhone 6s Plus und neuere iOS-Geräte benötigen mindestens iOS Version 13.7. Diese Versionen des iOS-Betriebssystems beinhalten auch die benötigte Version 2 des Apple Exposure Notification Framework.",
```

**Review by Ein-Tim — APPROVED** [review]
2 small things before merging (:

**Comment by Ein-Tim** [comment]
Okay, to late, but this was just a small thing, I think it's okay 👍

Thanks!

**Comment by dsarkar** [comment]
@Ein-Tim So sorry. Merges already.

**Comment by dsarkar** [comment]
maybe the more precise wording would be API 2. Need to look into it.

**Comment by Ein-Tim** [comment]
@dsarkar 
No problem at all! You are doing so great work here, and really a huge thanks for including the community like this! ❤️

**Comment by MikeMcC399** [comment]
@dsarkar 
https://www.coronawarn.app/en/faq/#minimum_requirements
https://www.coronawarn.app/de/faq/#minimum_requirements

[Apple/iOS]: Why is iOS 13.7 a minimum requirement? 
links to #no_risk_update
That doesn't look right.

The blog links are absolute, where they would be better as relative, but that is not so important.

**Comment by dsarkar** [comment]
@MikeMcC399 

> The blog links are absolute, where they would be better as relative, but that is not so important.

Thanks for the hint. Fixed.

**Comment by MikeMcC399** [comment]
@dsarkar 
>>The blog links are absolute, where they would be better as relative, but that is not so important.
>
>Thanks for the hint. Fixed.

Very good! It is noticeable if you test using npm because all the links should be going to the localhost copy not to the production website.

**Pull request merged**
